test/unit/authInfoTest.ts

```diff
@@ -2008,4 +2008,90 @@ describe('AuthInfo', () => {
       }
     });
   });
+  describe('Handle User Get Errors', () => {
+    let authCodeConfig;
+    beforeEach(async () => {
+      authCodeConfig = {
+        authCode: testMetadata.authCode,
+        loginUrl: testMetadata.loginUrl,
+      };
+      const authResponse = {
+        access_token: testMetadata.accessToken,
+        instance_url: testMetadata.instanceUrl,
+        id: '00DAuthInfoTest_orgId/005AuthInfoTest_userId',
+        refresh_token: testMetadata.refreshToken,
+      };
+
+      // Stub the http requests (OAuth2.requestToken() and the request for the username)
+      _postParmsStub.returns(Promise.resolve(authResponse));
+    });
+    it('user get returns 403 with body of json array', async () => {
+      const responseBody = {
+        statusCode: 403,
+        body: JSON.stringify([
+          {
+            message: 'The REST API is not enabled for this Organization',
+            errorCode: 'RESTAPINOTENABLED',
+          },
+        ]),
+      };
+      stubMethod($$.SANDBOX, Transport.prototype, 'httpRequest').returns(Promise.resolve(responseBody));
+      try {
+        await AuthInfo.create({ oauth2Options: authCodeConfig });
+      } catch (err) {
+        expect(err).to.have.property('message').to.include('The REST API is not enabled for this Organization');
+      }
+    });
+    it('user get returns 403 with body of json map', async () => {
+      const responseBody = {
+        statusCode: 403,
+        body: JSON.stringify({
+          message: 'The REST API is not enabled for this Organization',
+          errorCode: 'RESTAPINOTENABLED',
+        }),
+      };
+      stubMethod($$.SANDBOX, Transport.prototype, 'httpRequest').returns(Promise.resolve(responseBody));
+      try {
+        await AuthInfo.create({ oauth2Options: authCodeConfig });
```

Comment:
Ditto on all of these, you need to assert failure if you can't get the `.reject` to work.